core/src/test/scala/org/apache/spark/ui/UISeleniumSuite.scala

```diff
@@ -651,6 +671,15 @@ class UISeleniumSuite extends SparkFunSuite with WebBrowser with Matchers with B
     }
   }
 
+  def getResponseCode(url: URL, method: String): Int = {
+    val connection = url.openConnection().asInstanceOf[HttpURLConnection]
+    connection.setRequestMethod(method)
+    connection.connect()
+    val code = connection.getResponseCode()
+    connection.disconnect()
```

Comment:
It might just because it's late and I'm tired but I'm not quite sure where you think the try-finally should be